src/utilities/util_dump.c

```diff
@@ -904,9 +905,10 @@ dump_explore_bookmark_save(WT_CURSOR *cursor, char **bookmarks)
     for (i = 0; i < MAX_BOOKMARKS; ++i) {
         if (bookmarks[i] != NULL)
             continue;
-        if ((bookmarks[i] = malloc(strlen(key))) == NULL)
+        key_len = strlen(key);
+        if ((bookmarks[i] = malloc(key_len)) == NULL)
```

Comment:
As `strlen` does not include the space needed for the trailing NUL the `strncpy` below will truncate the key by one char.
```suggestion
       size_t key_size = strlen(key) + 1;
        if ((bookmarks[i] = malloc(key_size)) == NULL)
            /* handle error */
        memmove(bookmarks[i], key, key_size);
```

Reply:
Agreed, see 4648d0a489b77d5047b02260186c5a364131e409.